spec/tictactoe_spec.rb

```diff
@@ -1,26 +1,117 @@
+
+def is_row_outputed( io: , row:  )
+    expect( io.stdout.include? row).to eql true
+end
 
-     expect(game.renderBoard.length).to eql 9
-    end 
+describe TicTacToe do
 
-    it 'returns a list of all strings' do 
-     game = TicTacToe.new
+  context 'when the board is rendered' do
+    it 'returns a single list with 9 elements' do
+     game = TicTacToe.new(io: FakeIO.new)
+
+     expect(game.render_board.length).to eql 9
+    end
 
-     expect(game.renderBoard.all?{ |space| space.class == String && space.length <= 1 } ).to eql true 
-    end 
   end
 
-  context 'when the game starts the board is empty' do 
-    it 'sets the board to a list with empty strings' do 
-        game = TicTacToe.new
-        game.newGame 
-        expect( game.renderBoard.all?{ |space| space == '' } ).to eql true
-    end 
-  end   
+  context 'when the game starts the board is empty' do
+    it 'sets the board to a list with a single space' do
+        game = TicTacToe.new(io: FakeIO.new)
+        game.newGame
+        expect( game.render_board.all?{ |space| space == ' ' } ).to eql true
+    end
+  end
+
+ context 'when a single token is placed on an empty board' do
+    it 'the token will be "X"' do
+      game = TicTacToe.new(io: FakeIO.new)
+      expect( game.currentPlayer).to eql 'X'
+    end
+
+    it 'the specified space will place the current token on the board' do
+      fakeIO = FakeIO.new
+      game = TicTacToe.new(io: fakeIO)
+      game.placeToken(5)
+
+      is_row_outputed(io: fakeIO, row:" | |X")
+    end
+
+    it 'the token can be placed anywhere on the board' do
+      fakeIO = FakeIO.new
+      game = TicTacToe.new(io: fakeIO)
+
+      game.placeToken(3)
+      is_row_outputed(io: fakeIO, row:"X| | " )
+
+      game.newGame
+      fakeIO.stdout = []
+      game.placeToken(5)
+      is_row_outputed(io: fakeIO, row: " | |X" )
+
+
+      game.newGame
+      fakeIO.stdout = []
+      game.placeToken(7)
+      is_row_outputed(io: fakeIO, row: " |X| ")
+    end
+
+    it 'renders the board after a token is placed' do
+      fakeIO = FakeIO.new
+      game = TicTacToe.new(io: fakeIO)
+
+       game.placeToken(3)
+       expectedBoardOutput = [" | | ", "------", "X| | ", "------", " | | "]
+
+       expect( fakeIO.stdout ).to eql expectedBoardOutput
+    end
+  end
+
+  context 'when the user enters a desired position' do
+    it 'renders the board with their token' do
+      desired_position = 4
+
+      user_input = StringIO.new(desired_position.to_s)
+      fakeIO = FakeIO.new( stdin: user_input )
+      game = TicTacToe.new(io: fakeIO )
+
+      game.player_turn
+
+      is_row_outputed(io: fakeIO, row: " |X| ")
+    end
+  end
+
+ context 'when the user tries to place a token in a free space' do
+    it 'renders the board with their token in the specified space' do
+      desired_position = 4
+
+      user_input = StringIO.new(desired_position.to_s)
+      fakeIO = FakeIO.new( stdin: user_input )
+      game = TicTacToe.new(io: fakeIO )
```

Comment:
As mentioned in the previous comment, you could inline some of this with a simpler fake

```ruby
io = FakeIO.new(input: ['4'])
game = TicTacToe.new(io: io)
# ...
```